### PR TITLE
Google Fonts: update the method used to preconnect Fonts source domain.

### PR DESCRIPTION
**Pull request by jeherve**
Follow-up from #23045
Fixes #23306

#### Changes proposed in this Pull Request:

Instead of manually outputting a link to the `head`, let's rely on a Core filter, `wp_resource_hints`, to take care of that for us.
Let's take that opportunity to ensure we only do that when the Webfonts API is supported on the site.

#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* See #23045 for the basics.
* Enable the module and check that Google fonts are available.
* Enable and disable the Gutenberg plugin and make sure no errors are logged.

### CODE REVIEW

**Review by creativecoder — APPROVED** [review]
This works great. TIL about `wp_resource_hints`!

File: projects/plugins/jetpack/modules/google-fonts.php [hunk 1]
```diff
@@ -96,5 +96,4 @@ function jetpack_add_google_fonts_provider() {
 }
 add_action( 'after_setup_theme', 'jetpack_add_google_fonts_provider' );
 
-// Run on an early priority to print out the preconnect link tag near the start of the page source.
-add_action( 'wp_head', '\Automattic\Jetpack\Fonts\Google_Fonts_Provider::preconnect_font_source', 0 );
+add_filter( 'wp_resource_hints', '\Automattic\Jetpack\Fonts\Google_Fonts_Provider::font_source_resource_hint', 10, 2 );
```
Comment (creativecoder) [inline projects/plugins/jetpack/modules/google-fonts.php:99]: If Gutenberg is disabled, I'm seeing `PHP Warning:  Invalid argument supplied for foreach() in /Users/Grant/Automattic/Sites/wordpress-develop/src/wp-includes/general-template.php on line 3346`, as the `font_source_resource_hint` method is not available to hook into.

**Review by creativecoder — CHANGES_REQUESTED** [review]
Spoke too soon 🙂. This works great for adding the preconnect link, but we still need a check before adding the filter.

**Review by kraftbj — APPROVED** [review]

**Comment by creativecoder** [comment]
Sorry for the delay in me getting back to this... thanks for taking care of it!

**Pull request merged**